packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -289,6 +289,7 @@ export const DataGrid: React.FC<GridProps> = (props) => {
     [curPage, onPageChangeActionCallback],
   );
 
+  const namedDataString = JSON.stringify(namedData);
```

Comment:
Here `namedData` is a array we are getting from `useTemplateData` and we are passing this in the callback dependancies, and this array is generating new reference address on each render, so it cause the infinite render, because we are passing this method in `useRegisterBindings`.

so I converted this into JSON.stringify, the same we handle for `allowedExtension` in `usePickFiles`

Reply:
`useTemplateData` memoizes the array, it shouldn't return a new reference on each render.

Reply:
@sagardspeed2 please fix this in the `useTemplateData` hook like we discussed.